client.go

```diff
@@ -327,7 +327,12 @@ func (c *Client) acquireLock(ctx context.Context, opt *acquireLockOptions) (*Loc
 			return l, nil
 		}
 		c.logger.Println("Sleeping for a refresh period of ", getLockOptions.refreshPeriodDuration)
-		time.Sleep(getLockOptions.refreshPeriodDuration)
+		select {
+		case <-ctx.Done():
```

Comment:
`ctx.Err()` already does the trick - and if the context is canceled, the error in line 325 is going to make an early bail-out no matter what. I believe this change adds noise to the logic.

Reply:
If `cancel()` is called on the Context while the goroutine is sleeping via `time.Sleep()` then one must wait for the sleep to finish before the goroutine is aborted.

More practically, if I hit ctrl-c while waiting for a lock (and my SIGINT handler of course calls `cancel()`) I need to wait what's left of the refresh duration before the goroutine terminates, whereas with this change it terminates immediately.

And, unlike the other section, it's clearer this wouldn't affect correctness.

Reply:
To make my use case clearer, my refresh duration is actually 30 seconds, because I'm using this library for a leader election of a very long running process, so the common case is the lock is either acquired quickly (leader), or it's sitting there for days retrying (follower).  So if I want to abort the follower waiting for the lock, I have to wait up to 30 seconds for it to terminate cleanly.

Reply:
Please, remove the if-block in line 320.